keep-ui/app/mapping/create-new-mapping.tsx

```diff
@@ -16,31 +16,56 @@ import {
   Icon,
 } from "@tremor/react";
 import { useSession } from "next-auth/react";
-import { ChangeEvent, FormEvent, useMemo, useState } from "react";
+import {ChangeEvent, FormEvent, useEffect, useMemo, useRef, useState} from "react";
 import { usePapaParse } from "react-papaparse";
 import { toast } from "react-toastify";
 import { getApiURL } from "utils/apiUrl";
 import { useMappings } from "utils/hooks/useMappingRules";
+import {MappingRule} from "./models";
 
-export default function CreateNewMapping() {
+export default function CreateNewMapping( {editRule, editCallback}: {editRule: MappingRule | null; editCallback: (rule: MappingRule | null) => void } ) {
   const { data: session } = useSession();
   const { mutate } = useMappings();
   const [mapName, setMapName] = useState<string>("");
   const [fileName, setFileName] = useState<string>("");
   const [mapDescription, setMapDescription] = useState<string>("");
   const [selectedAttributes, setSelectedAttributes] = useState<string[]>([]);
   const [priority, setPriority] = useState<number>(0);
+  const [editMode, setEditMode] = useState(false);
+  const inputFile = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (editRule !== null) {
+      handleFileReset();
+      setEditMode(true);
+      setMapName(editRule.name);
+      setFileName(editRule.file_name? editRule.file_name : "");
+      setMapDescription(editRule.description ? editRule.description : "");
+      setSelectedAttributes(editRule.attributes ? editRule.attributes : []);
+      setPriority(editRule.priority);
+    }
+  }, [editRule]);
```

Comment:
Why is that needed? Add documentation

Reply:
Once the user clicks the edit button on the right hand side table, the form on the left hand side gets populated with all the information, the above snippet populates those fields.